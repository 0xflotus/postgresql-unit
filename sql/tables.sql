SET client_min_messages = warning;

DROP TABLE IF EXISTS prefixes;
CREATE TABLE prefixes (p text);
DELETE FROM prefixes;
INSERT INTO prefixes VALUES
	('y'),('z'),('a'),('f'),('p'),('n'),('µ'),('mu'),('m'),('c'),('d'),
	(''),
	('da'),('h'),('k'),('M'),('G'),('T'),('P'),('E'),('Z'),('Y');

DROP TABLE IF EXISTS factors;
CREATE TABLE factors (f double precision);
DELETE FROM factors;
INSERT INTO factors VALUES
	(1e27), (1e24), (1e21), (1e18), (1e15), (1e12), (1e9), (1e6), (1e3),
	(1),
	(1e-3), (1e-6), (1e-9), (1e-12), (1e-15), (1e-18), (1e-21), (1e-24), (1e-27), (1e-30),
	(0);

DROP TABLE IF EXISTS units;
CREATE TABLE units (u text, unit unit, base boolean, coherent boolean, duplicate boolean);
COMMENT ON COLUMN units.duplicate IS 'For coherent units, this unit has the same dimension as some other base or coherent unit';
DELETE FROM units;
COPY units (u, base, coherent, duplicate) FROM STDIN;
m/m	t	f	f
m	t	f	f
kg	t	f	f
s	t	f	f
A	t	f	f
K	t	f	f
mol	t	f	f
cd	t	f	f
B	t	f	f
sr	f	t	t
Hz	f	t	f
N	f	t	f
Pa	f	t	f
J	f	t	f
W	f	t	f
C	f	t	f
V	f	t	f
F	f	t	f
Ω	f	t	f
ohm	f	t	t
S	f	t	f
Wb	f	t	f
T	f	t	f
H	f	t	f
°C	f	t	t
lm	f	t	t
lx	f	t	f
Bq	f	t	t
Gy	f	t	f
rad	f	f	t
Sv	f	t	t
kat	f	t	f
g	f	f	t
a	f	f	f
l	f	f	f
t	f	f	f
bar	f	f	f
min	f	f	f
h	f	f	f
d	f	f	f
in	f	f	f
ft	f	f	f
yd	f	f	f
mi	f	f	f
oz	f	f	f
lb	f	f	f
\.
UPDATE units
	SET unit = u::unit;

SELECT * FROM units;
