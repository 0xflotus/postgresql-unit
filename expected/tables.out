SET client_min_messages = warning;
DROP TABLE IF EXISTS prefixes;
CREATE TABLE prefixes (p text);
DELETE FROM prefixes;
INSERT INTO prefixes VALUES
	('y'),('z'),('a'),('f'),('p'),('n'),('µ'),('mu'),('m'),('c'),('d'),
	(''),
	('da'),('h'),('k'),('M'),('G'),('T'),('P'),('E'),('Z'),('Y');
DROP TABLE IF EXISTS factors;
CREATE TABLE factors (f double precision);
DELETE FROM factors;
INSERT INTO factors VALUES
	(1e27), (1e24), (1e21), (1e18), (1e15), (1e12), (1e9), (1e6), (1e3),
	(1),
	(1e-3), (1e-6), (1e-9), (1e-12), (1e-15), (1e-18), (1e-21), (1e-24), (1e-27), (1e-30),
	(0);
DROP TABLE IF EXISTS units;
CREATE TABLE units (u text, unit unit, base boolean, coherent boolean, duplicate boolean);
COMMENT ON COLUMN units.duplicate IS 'For coherent units, this unit has the same dimension as some other base or coherent unit';
DELETE FROM units;
COPY units (u, base, coherent, duplicate) FROM STDIN;
UPDATE units
	SET unit = u::unit;
SELECT * FROM units;
  u  |      unit      | base | coherent | duplicate 
-----+----------------+------+----------+-----------
 m/m | 1              | t    | f        | f
 m   | 1 m            | t    | f        | f
 kg  | 1 kg           | t    | f        | f
 s   | 1 s            | t    | f        | f
 A   | 1 A            | t    | f        | f
 K   | 1 K            | t    | f        | f
 mol | 1 mol          | t    | f        | f
 cd  | 1 cd           | t    | f        | f
 B   | 1 B            | t    | f        | f
 sr  | 1              | f    | t        | t
 Hz  | 1 Hz           | f    | t        | f
 N   | 1 N            | f    | t        | f
 Pa  | 1 Pa           | f    | t        | f
 J   | 1 J            | f    | t        | f
 W   | 1 W            | f    | t        | f
 C   | 1 C            | f    | t        | f
 V   | 1 V            | f    | t        | f
 F   | 1 F            | f    | t        | f
 Ω   | 1 Ω            | f    | t        | f
 ohm | 1 Ω            | f    | t        | t
 S   | 1 S            | f    | t        | f
 Wb  | 1 Wb           | f    | t        | f
 T   | 1 T            | f    | t        | f
 H   | 1 H            | f    | t        | f
 °C  | 1 K            | f    | t        | t
 lm  | 1 cd           | f    | t        | t
 lx  | 1 lx           | f    | t        | f
 Bq  | 1 Hz           | f    | t        | t
 Gy  | 1 Gy           | f    | t        | f
 rad | 10 mGy         | f    | f        | t
 Sv  | 1 Gy           | f    | t        | t
 kat | 1 kat          | f    | t        | f
 g   | 1 g            | f    | f        | t
 a   | 100 m^2        | f    | f        | f
 l   | 0.001 m^3      | f    | f        | f
 t   | 1 Mg           | f    | f        | f
 bar | 100 kPa        | f    | f        | f
 min | 60 s           | f    | f        | f
 h   | 3.6 ks         | f    | f        | f
 d   | 86.4 ks        | f    | f        | f
 in  | 25.4 mm        | f    | f        | f
 ft  | 304.8 mm       | f    | f        | f
 yd  | 914.4 mm       | f    | f        | f
 mi  | 1.609344 km    | f    | f        | f
 oz  | 28.349523125 g | f    | f        | f
 lb  | 453.59237 g    | f    | f        | f
(46 rows)

